client/termdb/TermdbVocab.js

```diff
@@ -729,12 +729,14 @@ export class TermdbVocab extends Vocab {
 
 		let numResponses = 0
 		if (opts.loadingDiv) opts.loadingDiv.html('Updating data ...')
+		let termsPerRequest = opts.termsPerRequest || 1
 		// fetch the annotated sample for each term
-		while (termsToUpdate.length) {
+		let index = 0
+		while (index < termsToUpdate.length) {
+			const tws = termsToUpdate.slice(index, index + termsPerRequest)
 			// request data for one term each time, empty list and break while loop
 			// possible to change to pop two or more each time
-			const tw = termsToUpdate.pop()
-			const copy = this.getTwMinCopy(tw)
+			const copies = tws.map(tw => this.getTwMinCopy(tw))
```

Comment:
can be simplified to below, since `getTwMinCopy()` does not use the `this` keyword in it so can directly use that function as a callback
```js
const copies = tws.map(this.getTwMinCopy)
```